string map_pgm_str
---
string confirm_update
